src/components/AddUsersModal/index.jsx

```diff
@@ -0,0 +1,297 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Field, reduxForm, SubmissionError } from 'redux-form';
+import { Button, Icon, Modal } from '@edx/paragon';
+
+import H3 from '../H3';
+import TextAreaAutoSize from '../TextAreaAutoSize';
+import StatusAlert from '../StatusAlert';
+import FileInput from '../FileInput';
+import {
+  validateEmailTemplateFields,
+  validateEmailAddresses,
+  validateEmailAddressesFields,
+  mergeErrors,
+} from '../../utils';
+
+class AddUsersModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.errorMessageRef = React.createRef();
+    this.modalRef = React.createRef();
+
+    this.validateFormData = this.validateFormData.bind(this);
+    this.handleModalSubmit = this.handleModalSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const {
+      enterpriseId,
+      fetchSubscriptionDetails,
+    } = this.props;
+
+    if (enterpriseId) {
+      fetchSubscriptionDetails(enterpriseId);
+    }
+
+    const { current: { firstFocusableElement } } = this.modalRef;
+
+    if (firstFocusableElement) {
+      firstFocusableElement.focus();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      submitFailed,
+      submitSucceeded,
+      onClose,
+      error,
+    } = this.props;
+
+    const errorMessageRef = this.errorMessageRef && this.errorMessageRef.current;
+
+    if (submitSucceeded && submitSucceeded !== prevProps.submitSucceeded) {
+      onClose();
+    }
+
+    if (submitFailed && error !== prevProps.error && errorMessageRef) {
+      // When there is an new error, focus on the error message status alert
+      errorMessageRef.focus();
+    }
+  }
+
+  componentWillUnmount() {
+    // Clear the subscription data
+    this.props.clearSubscriptionDetails();
```

Comment:
Similar question around why we have this in the modal. If the "Subscription Management" page fetches the data at initial page load, this means we already the data stored in Redux or the `SubscriptionContext`.

If we clear out that data, wouldn't that impact the rest of the page?

Reply:
I have removed this, I am now using the `SubscriptionContext` and passing the required data as props.